Objects/dictobject.c

```diff
@@ -2871,8 +2735,8 @@ PyDict_SetDefault(PyObject *d, PyObject *key, PyObject *defaultobj)
             if (insertion_resize(mp) < 0) {
                 return NULL;
             }
-            hashpos = find_empty_slot(mp->ma_keys, key, hash);
         }
+        Py_ssize_t hashpos = find_empty_slot(mp->ma_keys, hash);
```

Comment:
Yes, from Python 3.6.
While this make hard to backporting future commits to Python 3.5,
Python 3.5 will go security-fix mode soon.